lib/extension/dlisio/ext/io.hpp

```diff
@@ -25,61 +34,42 @@ struct record {
 
 class stream {
 public:
-    explicit stream( const std::string& path ) noexcept (false);
+    explicit stream( lfp_protocol* f) noexcept (false);
 
-    record  at( int i ) noexcept (false);
-    record& at( int i, record& ) noexcept (false);
+    void close();
+    int eof() const noexcept (true);
 
-    void reindex( const std::vector< long long >&,
-                  const std::vector< int >& )
-        noexcept (false);
+    void seek( std::int64_t offset  ) noexcept (false);
 
-    void close();
+    std::int64_t tell() const noexcept(false);
+    std::int64_t absolute_tell() const noexcept(false);
 
-    void read( char* dst, long long offset, int n );
+    std::int64_t read( char* dst, int n ) noexcept (false);
+    std::int64_t read( char* dst, std::int64_t offset, int n ) noexcept (false);
 
 private:
-    std::fstream fs;
-    std::vector< long long > tells;
-    std::vector< int > residuals;
-
-    /*
-     * if this is true, there are no gaps inbetween tells, i.e. the file
-     * pointer should be at the next tell after reading. When stream is indexed
-     * with custom offsets, e.g. to account for broken files, this must be
-     * disabled and the safety check is gone
-     */
-    bool contiguous = true;
+    lfp_protocol* protocol;
 };
 
 
 struct stream_offsets {
-    /*see dlis_index_records. only change is tells.
-    tells here mean positive distance from the beginning of the file*/
-    std::vector< long long > tells;
-    std::vector< int > residuals;
-    std::vector< int > explicits;
-
-    void resize( std::size_t ) noexcept (false);
+    std::vector< long long > explicits;
+    std::vector< long long > implicits;
 };
 
-void map_source( mio::mmap_source&, const std::string& ) noexcept (false);
-void unmap( mio::mmap_source& ) noexcept (false);
+stream open(const std::string&, std::int64_t, bool, bool) noexcept (false);
 
-long long findsul( mio::mmap_source& file ) noexcept (false);
-long long findvrl( mio::mmap_source& path, long long from ) noexcept (false);
+long long findsul( stream& file ) noexcept (false);
+long long findvrl( stream& path, long long from ) noexcept (false);
 
-stream_offsets findoffsets( mio::mmap_source& path,
-                            long long from )
-noexcept (false);
+stream_offsets findoffsets( dl::stream& file ) noexcept (false);
 
 std::vector< std::pair< std::string, int > >
```

Comment:
As our 3.4GB file proved, you want here long long instead of int. Or tells will suddenly become negative.
You would also need to change parameter type in read_fdata in core.cpp